controllers/kserve_inferenceservice_controller_test.go

```diff
@@ -95,6 +103,89 @@ var _ = Describe("The Openshift Kserve model controller", func() {
+
+			err = cli.Status().Update(ctx, deployedInferenceService)
+			Expect(err).NotTo(HaveOccurred())
+
+			isvcService, err := waitForService(cli, testNs, inferenceService.Name, 5, 2*time.Second)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(isvcService.Annotations[constants.ServingCertAnnotationKey]).Should(Equal(inferenceService.Name))
+		})
+
+		It("should create a secret for runtime and update kserve local gateway in the istio-system namespace", func() {
+			// Stub: Create a kserve-local-gateway, which must be created by the OpenDataHub operator.
+			kserveLocalGateway := &istioclientv1beta1.Gateway{}
+			err := convertToStructuredResource(kserveLocalGatewayPath, kserveLocalGateway)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cli.Create(ctx, kserveLocalGateway)).Should(Succeed())
+
+			// Create a new InferenceService
+			inferenceService := &kservev1beta1.InferenceService{}
+			err = convertToStructuredResource(KserveInferenceServicePath1, inferenceService)
+			Expect(err).NotTo(HaveOccurred())
+			inferenceService.SetNamespace(testNs)
+
+			Expect(cli.Create(ctx, inferenceService)).Should(Succeed())
+
+			// Update the URL of the InferenceService to indicate it is ready.
+			deployedInferenceService := &kservev1beta1.InferenceService{}
+			err = cli.Get(ctx, types.NamespacedName{Name: inferenceService.Name, Namespace: inferenceService.Namespace}, deployedInferenceService)
+			Expect(err).NotTo(HaveOccurred())
+
+			url, err := apis.ParseURL("https://example-onnx-mnist-default.test.com")
+			Expect(err).NotTo(HaveOccurred())
+			deployedInferenceService.Status.URL = url
+
+			err = cli.Status().Update(ctx, deployedInferenceService)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Stub: Create a certificate Secret, which must be created by the openshift service-ca operator.
+			secret := &corev1.Secret{}
+			err = convertToStructuredResource(testIsvcSvcSecretPath, secret)
+			Expect(err).NotTo(HaveOccurred())
+			secret.SetNamespace(inferenceService.Namespace)
+			Expect(cli.Create(ctx, secret)).Should(Succeed())
+
+			// Verify that the certificate secret is found in the same namespace where the InferenceService is created.
+			_, err = waitForSecret(cli, inferenceService.Namespace, inferenceService.Name, 5, 1*time.Second)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Verify that the certificate secret is created in the istio-system namespace.
+			_, err = waitForSecret(cli, constants.IstioNamespace, inferenceService.Name, 5, 5*time.Second)
+			Expect(err).NotTo(HaveOccurred())
+
+			gateway, err := waitForUpdatedGatewayCompletion(cli, "add", constants.IstioNamespace, constants.KServeGatewayName, inferenceService.Name, 5, 2*time.Second)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Ensure that the server is successfully added to the KServe local gateway within the istio-system namespace.
+			targetServerExist := isServerExistFromGateway(gateway, inferenceService.Name)
+			Expect(targetServerExist).Should(BeTrue())
```

Comment:
Again... it should be better to use `Eventually` for these checks.

Reply:
updated